### ISSUE TITLE
Add SpeedHack Audio

### ISSUE THREAD

**Issue by cheznewa** [iArtorias/speedhack-cpp#1]
hello, this project is super i waiting one it!, and is possible to add speedhack but with audio pitch?

thanks, Newa Our Master.

**Comment by iArtorias** [comment]
Hi, the idea is interesting. I will think about it.

**Comment by Trxmor** [comment]
Hi iArtoias,
I compare other couple of similar works but I prefer yours because of you meticulous coding and the stub. Do you mind add a keyboard shortcut? Thanks!

**Comment by Trxmor** [comment]
I tried this but it doesn't work
    CSpeedHack() 
    {
        // Try reading the 'speedhack.txt' to obtain the speed multiplier for our speedhack
        std::wfstream stream( CProxyStub::instance().m_proxy_path / "speedhack.txt" );

        if (stream.good() || !stream.fail())
        {
            std::wstring _speed;
            std::getline( stream, _speed );

            if (!_speed.empty())
            {
                try
                {
                    size_t ldx;
                    auto speed = std::stod( _speed, &ldx );

                    m_speed_multiplier = speed;

                    if (GetAsyncKeyState(VK_DOWN) & 1) {
                        speed = 3.0;
                    }
                    if (GetAsyncKeyState(VK_UP) & 1) {
                        m_speed_multiplier = 3.0;
                    }
                }
                catch (const std::invalid_argument&)
                {}
            }
        }
    };

**Comment by iArtorias** [comment]
Hi, I can add it since it was in my to-do list anyway. The reason why your code isn't working is because you only run it once upon the function initialization.
TL;DR you have to create a separate running thread where you can constantly check for the pressed key event.

**Comment by Trxmor** [comment]
> Hi, I can add it since it was in my to-do list anyway. The reason why your code isn't working is because you only run it once upon the function initialization. TL;DR you have to create a separate running thread where you can constantly check for the pressed key event.

Thank you so much for replying, iArtorias!

However, this is beyond my knowledge and have little clue of what you're saying. I'm currently at a learning stage, a noob.

**Comment by Trxmor** [comment]
I could go on others that have the keyboard feature. It works but of course your works portray you mindset and your are simply unique & creative to me. And I really doubt when I see their coding styles not mentioning compilation errors and the fact, it's obvious have no originality or ripped off upon ripped off snippets.

Please create one or at least a snippet for newbie like me. Thank you so much!

**Comment by divinity76** [comment]
I had limited success speedhacking StarCraft2 Wings of Liberty with
 IAudioClockAdjustment::SetSampleRate
- it worked perfectly on cutscenes!
- gameplay background music is annoying at the wrong speed
- [Marines](https://starcraft.fandom.com/wiki/Marine) get child-voices at 1.6x speed.
- [Dr Hanson](https://starcraft.fandom.com/wiki/Ariel_Hanson) get a super deep-male voice at 0.7x speed (Hanson is female)

Ultimately, the annoying background music effect made it not worthwhile, I removed the SetSampleRate experiment.
